# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/wmf/WMFVideoMFTManager.cpp
# Commit: a0d1aa84dd7e
# Full Hash: a0d1aa84dd7e1b861630c6cccb2094ad6d2a4297
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2024-07-25 09:31:33
# Description:
#   Bug 1909610 - Add mDXVA2Manager check in WMFVideoMFTManager::UseZeroCopyVideoFrame() r=media-playback-reviewers,alwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D217655
# ==============================================================================

diff -r 80bfb01e76a5 -r a0d1aa84dd7e dom/media/platforms/wmf/WMFVideoMFTManager.cpp
--- a/dom/media/platforms/wmf/WMFVideoMFTManager.cpp	Thu Jul 25 00:41:49 2024 +0000
+++ b/dom/media/platforms/wmf/WMFVideoMFTManager.cpp	Thu Jul 25 01:23:10 2024 +0000
@@ -1048,7 +1048,8 @@
 }
 
 bool WMFVideoMFTManager::UseZeroCopyVideoFrame() const {
-  if (mZeroCopyNV12Texture && mDXVA2Manager->SupportsZeroCopyNV12Texture()) {
+  if (mZeroCopyNV12Texture && mDXVA2Manager &&
+      mDXVA2Manager->SupportsZeroCopyNV12Texture()) {
     return true;
   }
   return false;
